py3status/modules/khal_calendar.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+"""
+Displays upcoming khal events.
+
+Configuration parameters:
+    cache_timeout: refresh interval for this module (default 60)
+    config_path: Path to khal configuration file. The default None resolves to /home/$USER/.config/khal/config (default None)
+    date_end: Until which datetime the module searches for events (default 'eod')
+    format: display format for this module (default '{appointments}')
+    output_format: khal conform format for displaying event output (default '{start-time} {title}')
```

Comment:
Things to do...

* Config `output_format` have `{start-time} {title}` placeholders. I don't see any other placeholders. If there is more possible placeholders, please list them too.
* Config `output_format` config should be renamed to `format_event` with same value.
* Config `format` should have `{format_event}` instead of `{appointments}`.
* Make new config `format_separator` or `format_event_separator` with `' '` value allowing users to configure their separator instead of hardcoded `' '.join()`. 
* Code would look similar to `hddtemp`, `nvidia_smi`, or modules containing `format*separator`.

Reply:
>     * Config `output_format` have `{start-time} {title}` placeholders. I don't see any other placeholders. If there is more possible placeholders, please list them too.
You can use all placeholders that are supported by `khal` itself. You can take a look at https://khal.readthedocs.io/en/latest/usage.html under `--format FORMAT` to get an idea.
So this option is more dependent on khal than on this module.
>     * Config `output_format` config should be renamed to `format_event` with same value.
Done.
>     * Config `format` should have `{format_event}` instead of `{appointments}`.
Done.
>     * Make new config `format_separator` or `format_event_separator` with `' '` value allowing users to configure their separator instead of hardcoded `' '.join()`.
>     * Use `composite_join` for separators instead of `' '.join()`.
Will work on that.